base/linalg/exceptions.jl

```diff
@@ -29,10 +29,30 @@ function Base.showerror(io::IO, ex::ARPACKException)
     end
 end
 
+"""
+   SingularException(info)
+
+The matrix passed is a singular matrix. Optional argument INFO is an INTEGER > 0  
+      if INFO = i, The matrix is exactly singular, and division by zero 
+      will occur if it is used to solve a system of equations.
+"""
 struct SingularException <: Exception
     info::BlasInt
 end
 
+function Base.showerror(io::IO, ex::SingularException)
```

Comment:
We'd need to add similar information here about the case where pivoting is disabled.